x-pack/platform/plugins/shared/cases/server/client/alerts/get.ts

```diff
@@ -37,3 +42,41 @@ export const getAlerts = async (
     ...alert._source,
   }));
 };
+
+export const getAlertMetadataFromComments = async (
+  comments: AttachmentAttributes[],
+  clientArgs: CasesClientArgs
+): Promise<CaseMetadata> => {
+  const alertInfo: AlertInfo[] = comments.flatMap((c) => {
+    if (c.type === AttachmentType.alert) {
+      const { ids, indices } = getIDsAndIndicesAsArrays(c);
+      return ids.map((alertId, index) => ({ id: alertId, index: indices[index] }));
+    }
+    return [];
+  });
+
+  const alertsDocs = await getAlerts(alertInfo, clientArgs);
+
+  const metadata = alertsDocs.reduce<Record<string, Set<string>>>(
```

Comment:
Technically nothing wrong with this, but although it doesn't read as well, I tend to prefer `for loops` when using nested looping constructs like this, because they tend to be faster. Currently, there's a cap on the number of alerts that can be attached to a case, so this wouldn't be problematic at all right now, but it could be if that cap significantly increases

Reply:
Replace the `reduce` with a `for loop` [here](https://github.com/elastic/kibana/pull/230192/commits/ae83d75eaa71cc4c88788f64357635a1082ff439).